blas/tpls/KokkosBlas1_nrm1_tpl_spec_decl.hpp

```diff
@@ -207,40 +111,65 @@ KOKKOSBLAS1_CNRM1_TPL_SPEC_DECL_BLAS(Kokkos::LayoutLeft, Kokkos::HostSpace,
     using execution_space = EXECSPACE;                                         \
-    typedef Kokkos::View<double, LAYOUT, Kokkos::HostSpace,                    \
-                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >             \
-        RV;                                                                    \
-    typedef Kokkos::View<const double*, LAYOUT,                                \
-                         Kokkos::Device<EXECSPACE, MEMSPACE>,                  \
-                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >             \
-        XV;                                                                    \
-    typedef typename XV::size_type size_type;                                  \
+    using RV = Kokkos::View<typename Kokkos::ArithTraits<SCALAR>::mag_type,    \
+                            LAYOUT, Kokkos::HostSpace,                         \
+                            Kokkos::MemoryTraits<Kokkos::Unmanaged>>;          \
+    using XV = Kokkos::View<const SCALAR*, LAYOUT,                             \
+                            Kokkos::Device<EXECSPACE, MEMSPACE>,               \
+                            Kokkos::MemoryTraits<Kokkos::Unmanaged>>;          \
+    using size_type = typename XV::size_type;                                  \
                                                                                \
     static void nrm1(const execution_space& space, RV& R, const XV& X) {       \
-      Kokkos::Profiling::pushRegion("KokkosBlas::nrm1[TPL_CUBLAS,double]");    \
+      Kokkos::Profiling::pushRegion("KokkosBlas::nrm1[TPL_CUBLAS," #SCALAR     \
+                                    "]");                                      \
       const size_type numElems = X.extent(0);                                  \
       if (numElems < static_cast<size_type>(INT_MAX)) {                        \
-        nrm1_print_specialization<RV, XV>();                                   \
-        const int N       = static_cast<int>(numElems);                        \
-        constexpr int one = 1;                                                 \
-        KokkosBlas::Impl::CudaBlasSingleton& s =                               \
-            KokkosBlas::Impl::CudaBlasSingleton::singleton();                  \
-        KOKKOS_CUBLAS_SAFE_CALL_IMPL(                                          \
-            cublasSetStream(s.handle, space.cuda_stream()));                   \
-        KOKKOS_CUBLAS_SAFE_CALL_IMPL(                                          \
-            cublasDasum(s.handle, N, X.data(), one, R.data()));                \
-        KOKKOS_CUBLAS_SAFE_CALL_IMPL(cublasSetStream(s.handle, NULL));         \
+        cublasAsumWrapper(space, R, X);                                        \
       } else {                                                                 \
         Nrm1<execution_space, RV, XV, 1, false, ETI_SPEC_AVAIL>::nrm1(space,   \
```

Comment:
Plugging in ETI_SPEC_AVAIL as both true and false here is necessarily re-instantiating the actual kernel once for each type combo, so fixing that will definitely be nice for build times later on...